### DISCUSSION TITLE
Terminal output grouping (GitHub Actions)

### DISCUSSION THREAD

**Discussion by amezin** [pytest-dev/pytest#11246] [Ideas]
GitHub Actions allow grouping console output by echo'ing special commands: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines

With current API I have no idea how to implement this.

Maybe there should be hooks that are called by the terminal reporter where it currently outputs a horizontal line (before and after the summary, before and after the summary, etc)?

**Comment by The-Compiler** [comment]
pytest-travis-fold [implements this](https://github.com/abusalimov/pytest-travis-fold/blob/master/pytest_travis_fold.py) by monkey patching the stock terminal reporter.

Adding some hooks to enable this kind of thing would definitely be nice IMHO though!

Also see https://github.com/abusalimov/pytest-travis-fold/issues/9 - as mentioned there, a `pytest-github-actions` plugin with other useful GHA features would be nice to have!
